lint/package.go

```diff
@@ -182,17 +183,41 @@ func (p *Package) scanSortable() {
 	}
 }
 
-func (p *Package) lint(rules []Rule, config Config, failures chan Failure) {
+func (p *Package) lint(rules []Rule, config Config, failures chan Failure) error {
 	p.scanSortable()
 	var wg sync.WaitGroup
+	errCh := make(chan error, 1) // Buffered channel to signal the first error
+    _, cancel := context.WithCancel(context.Background())
+
+    defer cancel() // Ensure the context is canceled at the end
+
 	for _, file := range p.files {
 		wg.Add(1)
 		go (func(file *File) {
-			file.lint(rules, config, failures)
-			wg.Done()
+			defer wg.Done()
+			if err := file.lint(rules, config, failures); err != nil {
+                select {
+                case errCh <- err: // Send the error to the channel if it's empty
+                    cancel() // Signal other goroutines to stop
```

Comment:
See [previous comment](https://github.com/mgechev/revive/pull/1126#discussion_r1844984367)


```suggestion
```

Reply:
reverted to original version, any suggestion how to handle error for ```file.lint(rules, config, failures)``` in proper way?

especialy if that code runs in go routine?